### PR TITLE
Scenarios Articles

### PR DESCRIPTION
**Pull request by pytlesk4**
- overview
- pass data between steps

Those are the two articles that are ready for review right now, going to add the rest to this pull request.

### CODE REVIEW

File: articles/testing/pass-data-steps.md [hunk 1]
```diff
@@ -1 +1,72 @@
...
+2. GET /todos/{todoId}
+3. PUT /todos/{todoId}
+4. DELETE /todos/{todoId}
+5. GET /todos/{todoId} (to make sure 404 not found)
+
+In order for this to work, steps 2-5 need to be able to access the `id` of the newly created todo in step 1. To do this, we'll save the new `id` property in step 1 to the $.ctx object.
+
+### The $.ctx object
+
+The $.ctx object makes it easy to store temporary data for use during the course of a single scenario run. It starts empty at the beginning of every run. Use it to share data between steps.
```
Comment (marbemac) [inline articles/testing/pass-data-steps.md]: Link to context variables article.

File: articles/testing/pass-data-steps.md [hunk 2]
```diff
@@ -1 +1,73 @@
...
+### When to use ctx variables
+
+You can think of ctx variables as a temporary state in your scenario. In general, we recommend using them for values that are created during the course of a single scenario that need to be shared with subsequent steps. This includes things like ids, usernames, and randomly generated tokens.
+
+<!-- theme: warning -->
+> We do **NOT** recommend using ctx variables for more permanent values that need to persist across scenario runs. For this persisted value use case, we recommend using the $$.env object, which is explained in detail [here](variables-environment.md).
+
+### Debugging ctx variables
+
+$.ctx variables can be manually set when running a step. This is useful if you are trying to debug a single step in the middle of a scenario that relies on a ctx value set in a previous step. Instead of running the entire scenario to generate the needed ctx value, you can set it in the UI manually. In the screenshot below, we can set the todoId ctx variable before running the step to debug what would happen with a particular todoId value.
```
Comment (rowa97) [inline articles/testing/pass-data-steps.md]: "In the screenshot below..." needs to be reworked, awkward framing

File: articles/testing/scenarios-introduction.md [hunk 3]
```diff
@@ -1,17 +1,26 @@
-Stoplight Scenarios is a powerful (but accessible!) tool that takes the pain out of API testing. It is a standalone product, available on [the web](https://scenarios.stoplight.io), and as a [desktop app](https://download-next.stoplight.io).
+Stoplight Scenarios is a powerful (but accessible!) tool that takes the pain out of API testing. It's available on [the web](https://next.stoplight.io) and as a [desktop app](https://download-next.stoplight.io).
 
 We generally recommend the desktop app when possible. It works with local servers, behind firewalls, and exchanges information with tools on your computer like Git or your favorite IDE. You can switch seamlessly between the desktop app and the web app.
```
Comment (rowa97) [inline articles/testing/scenarios-introduction.md]: needs to be updated

File: articles/testing/scenarios-introduction.md [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
+* **Flexible** Your APIs, your tests, your way. Scenarios only test what you want them to. They have no opinion about your architecture (Monolithic vs Microservices), company structure (in house vs distributed), development lifecycle (production vs TDD), and your environment (development vs staging vs production).
 
-- **Fast** Time can’t be slowed down, and we can’t give it back to you. Creating tests should be quick, and waiting for you tests to run shouldn’t feel like watching water boil. Scenarios are run concurrently for maximum speed - run hundreds of API requests and test assertions in seconds.
+* **Fast** Time can’t be slowed down, and we can’t give it back to you. Creating tests should be quick, and waiting for you tests to run shouldn’t feel like watching water boil. Scenarios are run concurrently for maximum speed - run hundreds of API requests and test assertions in seconds.
```
Comment (rossmcdonald) [inline articles/testing/scenarios-introduction.md]: Maybe add a comment here pointing the reader to next steps, like:

> To get started with Scenarios, see the Prism getting started guide here.

File: articles/testing/scenarios-introduction.md [hunk 5]
```diff
@@ -1,17 +1,26 @@
-Stoplight Scenarios is a powerful (but accessible!) tool that takes the pain out of API testing. It is a standalone product, available on [the web](https://scenarios.stoplight.io), and as a [desktop app](https://download-next.stoplight.io).
+Stoplight Scenarios is a powerful (but accessible!) tool that takes the pain out of API testing. It's available on [the web](https://next.stoplight.io) and as a [desktop app](https://download-next.stoplight.io).
```
Comment (rossmcdonald) [inline articles/testing/scenarios-introduction.md]: Is the purpose of this article to go over the app (previously referred to as Scenarios), or to go over scenarios (the spec)? If it's the latter, then I think the introduction should be updated to reflect what scenarios actually are (a specification used to facilitate API testing). 

From reading this (and the paragraph below), it makes it seem like scenarios is the web/desktop app.

Comment (rossmcdonald) [reply articles/testing/scenarios-introduction.md]: Should also mention Prism, noting it's the official scenario runner/conductor.

File: articles/testing/scenarios-introduction.md [hunk 6]
```diff
@@ -1,17 +1,26 @@
...
+* **Flexible** Your APIs, your tests, your way. Scenarios only test what you want them to. They have no opinion about your architecture (Monolithic vs Microservices), company structure (in house vs distributed), development lifecycle (production vs TDD), and your environment (development vs staging vs production).
 
-- **Fast** Time can’t be slowed down, and we can’t give it back to you. Creating tests should be quick, and waiting for you tests to run shouldn’t feel like watching water boil. Scenarios are run concurrently for maximum speed - run hundreds of API requests and test assertions in seconds.
+* **Fast** Time can’t be slowed down, and we can’t give it back to you. Creating tests should be quick, and waiting for you tests to run shouldn’t feel like watching water boil. Scenarios are run concurrently for maximum speed - run hundreds of API requests and test assertions in seconds.
 
 ### Editor UI Overview
 
-![](https://cdn.stoplight.io/help-portal/scenarios/scenario-editor-callout.png)
+![](../../assets/images/scenarios-editor-overview.png)
```
Comment (rossmcdonald) [inline articles/testing/scenarios-introduction.md]: I think the image should have arrows pointing to different things if we want this be an overview of the UI.

File: articles/testing/pass-data-steps.md (hunk 2 above)
Comment (pytlesk4) [reply articles/testing/pass-data-steps.md]: Removed screen shot for right now

**Pull request closed without merging**